x-pack/plugin/inference/qa/mixed-cluster/src/javaRestTest/java/org/elasticsearch/xpack/inference/qa/mixed/CohereServiceMixedIT.java

```diff
@@ -0,0 +1,287 @@
+        serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
+        assertThat(serviceSettings, hasEntry("embedding_type", "float"));
+
+        assertEmbeddingInference(oldClusterIdInt8, CohereEmbeddingType.BYTE);
+        assertEmbeddingInference(oldClusterIdFloat, CohereEmbeddingType.FLOAT);
+
+        {
+            final String upgradedClusterIdInt8 = "upgraded-cluster-embeddings-int8";
+
+            cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseByte()));
+            put(upgradedClusterIdInt8, embeddingConfigInt8(getUrl(cohereEmbeddingsServer)), TaskType.TEXT_EMBEDDING);
+
+            configs = (List<Map<String, Object>>) get(TaskType.TEXT_EMBEDDING, upgradedClusterIdInt8).get("endpoints");
+            serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
+            assertThat(serviceSettings, hasEntry("embedding_type", "byte")); // int8 rewritten to byte
+
+            assertEmbeddingInference(upgradedClusterIdInt8, CohereEmbeddingType.INT8);
+            delete(upgradedClusterIdInt8);
+        }
+        {
+            final String upgradedClusterIdFloat = "upgraded-cluster-embeddings-float";
+            cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseFloat()));
+            put(upgradedClusterIdFloat, embeddingConfigFloat(getUrl(cohereEmbeddingsServer)), TaskType.TEXT_EMBEDDING);
+
+            configs = (List<Map<String, Object>>) get(TaskType.TEXT_EMBEDDING, upgradedClusterIdFloat).get("endpoints");
+            serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
+            assertThat(serviceSettings, hasEntry("embedding_type", "float"));
+
+            assertEmbeddingInference(upgradedClusterIdFloat, CohereEmbeddingType.FLOAT);
+            delete(upgradedClusterIdFloat);
+        }
+
+        delete(oldClusterIdFloat);
+        delete(oldClusterIdInt8);
+
+    }
+
+    void assertEmbeddingInference(String inferenceId, CohereEmbeddingType type) throws IOException {
+        switch (type) {
+            case INT8:
+            case BYTE:
+                cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseByte()));
+                break;
+            case FLOAT:
+                cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseFloat()));
+        }
+
+        var inferenceMap = inference(inferenceId, TaskType.TEXT_EMBEDDING, "some text");
+        assertThat(inferenceMap.entrySet(), not(empty()));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testRerank() throws IOException {
+        var rerankSupported = bwcVersion.onOrAfter(Version.fromString(COHERE_RERANK_ADDED));
+        assumeTrue("Cohere rerank service added in " + COHERE_RERANK_ADDED, rerankSupported);
+
+        final String oldClusterId = "old-cluster-rerank";
```

Comment:
nit: there is no old cluster only a mixed cluster

```suggestion
        final String inferenceId = "mixed-cluster-cohere-rerank";
```